### PR TITLE
Adds the Form 24 schema

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
#262

### CODE REVIEW

File: schema/F24.json [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+            "if":{
+                "properties": {
+                    "form_type": {
+                        "const": "F24A"
+                    }
+                },
+                "required": ["form_type"]
+            },
+            "then":{
+                "required": ["form_type"]
```
Comment (mjtravers) [inline schema/F24.json:360]: Did you mean to make "original_amendment_date" the property for this conditional statement?

File: schema/F24.json [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        "original_amendment_date": {
+            "title": "ORIGINAL AMENDMENT DATE",
+            "description": "",
+            "type": [
+                "string",
+                "null"
+            ],
+            "minLength": 0,
+            "maxLength": 8,
+            "pattern": "^\\d{0,8}$",
```
Comment (mjtravers) [inline schema/F24.json:96]: We've been using this pattern for dates:

"type": "string",
            "minLength": 10,
            "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}$",

File: schema/F24.json [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+                "SAMPLE_DATA": 20120729,
+                "VALUE_REFERENCE": "YYYYMMDD",
+                "RULE_REFERENCE": "Use date of original report or of most recent amendment.",
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "committee_name": {
+            "title": "COMMITTEE NAME",
+            "description": "",
+            "type": "string",
```
Comment (mjtravers) [inline schema/F24.json:114]: As a "X (warning)" field is it required? If not, this "type" needs to be:

"type": [
                "string",
                "null"
            ],

File: schema/F24.json [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
+                "SAMPLE_DATA": null,
+                "VALUE_REFERENCE": null,
+                "RULE_REFERENCE": null,
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "street_1": {
+            "title": "STREET 1",
+            "description": "",
+            "type": "string",
```
Comment (mjtravers) [inline schema/F24.json:132]: As a "X (warning)" field is it required? If not, this "type" needs to be:

"type": [
                "string",
                "null"
            ],

File: schema/F24.json [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+                "SAMPLE_DATA": null,
+                "VALUE_REFERENCE": null,
+                "RULE_REFERENCE": null,
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "street_2": {
+            "title": "STREET 2",
+            "description": "",
+            "type": "string",
```
Comment (mjtravers) [inline schema/F24.json:150]: This is an optional field, so "type" needs to accept null:

"type": [
                "string",
                "null"
            ],

File: schema/F24.json [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+                "SAMPLE_DATA": null,
+                "VALUE_REFERENCE": null,
+                "RULE_REFERENCE": null,
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "city": {
+            "title": "CITY",
+            "description": "",
+            "type": "string",
```
Comment (mjtravers) [inline schema/F24.json:168]: As a "X (warning)" field is it required? If not, this "type" needs to be:

"type": [
                "string",
                "null"
            ],

**Review by mjtravers — CHANGES_REQUESTED** [review]
Requesting some changes to be made to the JSON file.

File: schema/F24.json [hunk 7]
```diff
@@ -0,0 +1,365 @@
...
+    "$schema": "https://json-schema.org/draft-07/schema#",
+    "$id": "https://github.com/fecgov/fecfile-validate/blob/main/schema/F24.json",
+    "version": "8.3.0.1",
+    "title": "FEC F24",
+    "description": "FORM 24 - 24 / 48 HOUR NOTICE OF INDEPENDENT EXPENDITURE",
+    "type": "object",
+    "required": [
+        "form_type",
+        "filer_committee_id_number",
+        "report_type_{24_48_hour}",
```
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/F24.json:11]: I had changed this from `report_type_{24_48_hour}` to just `report_type`, but I failed to change it in the required list...  Fixed :+1:

File: schema/F24.json [hunk 8]
```diff
@@ -0,0 +1,365 @@
...
+        "street_1",
+        "city",
+        "state",
+        "zip"
+    ],
+    "properties": {
+        "form_type": {
+            "title": "FORM TYPE",
+            "description": "",
+            "type": "string",
```
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/F24.json:27]: :+1:

File: schema/F24.json [hunk 9]
```diff
@@ -0,0 +1,365 @@
...
+                "SAMPLE_DATA": "F24N",
+                "VALUE_REFERENCE": "F24+[N|A]",
+                "RULE_REFERENCE": null,
+                "FIELD_FORM_ASSOCIATION": null
+            }
+        },
+        "filer_committee_id_number": {
+            "title": "FILER COMMITTEE ID NUMBER",
+            "description": "",
+            "type": "string",
```
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/F24.json:48]: :+1:

File: schema/F24.json (hunk 1 above)
Comment (Elaine-Krauss-TCG) [reply schema/F24.json:360]: :100:

File: schema/F24.json (hunk 3 above)
Comment (Elaine-Krauss-TCG) [reply schema/F24.json:114]: After discussing the "X (warning)" status with Gregg, he recommends that we assume that these fields are required.  I'm adding them to the required array and updating their minimum lengths.

**Review by mjtravers — APPROVED** [review]

**Pull request merged**